### PR TITLE
Add VPN configuration details for different cloud providers

### PR DESCRIPTION
**Pull request by afastronomer**
Add missed information regarding Astro supported cross-account and cross-network connectivity.

### CODE REVIEW

**Review by Igor9052A — APPROVED** [review]

File: astro/connect-aws.md [hunk 1]
```diff
@@ -294,6 +295,37 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
```
Comment (lzdanski) [inline astro/connect-aws.md:302]:
```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

File: astro/connect-aws.md [hunk 2]
```diff
@@ -294,6 +295,37 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
...
+
+:::
+
+Use this connectivity type for access to on-premises resources or inside onther cloud providers.
```
Comment (lzdanski) [inline astro/connect-aws.md:306]:
```suggestion
Use this connectivity type to access on-premises resources or resources in other cloud providers.
```

File: astro/connect-aws.md [hunk 3]
```diff
@@ -294,6 +295,37 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
...
+- An Astro cluster
+- Configured Gateway device or Application with Public IP address (or 2 addresses for HA setup)
+
+2. Contact internal Network team or engineer and ask for the following information:
+
+- Public IP addresses for the tunnels configuration.
+- IKE pre-shared key, if your team wants to use a particular one.
+- Preferable settings for phase 1 and phase 2 (BGP only) IKE negotionations.
+- ASN for BGP or IP Prefixes for Static configuration.
+- Optional. A size /30 IPv4 CIDR block from the 169.254.0.0/16 range for the inside tunnel IPv4 addresses.
```
Comment (lzdanski) [inline astro/connect-aws.md:321]:
```suggestion

- An Astro Deployment with a dedicated cluster.
- Configured gateway device or application with Public IP address. You need 2 addresses for an HA setup.

Contact your internal network team or engineer and ask for the following information:

- Public IP addresses for the tunnels configuration.
- IKE pre-shared key, if your team wants to use a particular key.
- Preferable settings for phase 1 and phase 2 (BGP only) IKE negotiations.
- ASN for BGP or IP prefixes for static configuration.
- (Optional) A size /30 IPv4 CIDR block from the 169.254.0.0/16 range for the inside tunnel IPv4 addresses.
```

File: astro/connect-aws.md [hunk 4]
```diff
@@ -294,6 +295,37 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
...
+
+#### Contact Astronomer support for VPN configuration on Astro side
+
+Submit all collected details via Support portal. CRE team will proceed with the required steps. In case of additional questions or clarifications, CRE team will contact you within Support request or let you know about connectivity tests.
```
Comment (lzdanski) [inline astro/connect-aws.md:325]:
```suggestion
Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). The Astronomer CRE team will proceed with the required steps. The CRE team will contact you using your support ticket to ask follow-up questions, request clarification, or let you know about connectivity tests.
```

File: astro/connect-gcp.md [hunk 5]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
...
+Retrieve the following information about your VPN device or application:
+
+    - Public IP address
+    - Subnet CIDR (multipple if needed) on your side of the connection
+    - Preferences regarding shared key and BGP usage
+    - IKE settings for the tunnel
+
+#### Contact Astronomer support for VPN configuration on Astro side
+
+Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). CRE team will proceed with the required steps. In case of additional questions or clarifications, CRE team will contact you within Support request or let you know about connectivity tests.
```
Comment (lzdanski) [inline astro/connect-gcp.md:144]:
```suggestion
Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). The Astronomer CRE team will proceed with the required steps. The CRE team will contact you using your support ticket to ask follow-up questions, request clarification, or let you know about connectivity tests.
```

File: astro/connect-gcp.md [hunk 6]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is available only for dedicated Astro Hosted clusters and Astro Hybrid.
```
Comment (lzdanski) [inline astro/connect-gcp.md:127]:
```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

File: astro/connect-gcp.md [hunk 7]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
...
+:::
+
+Use this connectivity type for access to on-premises resources or inside onther cloud providers.
+
+#### Perequisites
+
+Retrieve the following information about your VPN device or application:
+
+    - Public IP address
+    - Subnet CIDR (multipple if needed) on your side of the connection
```
Comment (lzdanski) [inline astro/connect-gcp.md:138]:
```suggestion
    - Subnet CIDR range, multiple if needed, for your side of the connection
```

File: astro/connect-gcp.md [hunk 8]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
...
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is available only for dedicated Astro Hosted clusters and Astro Hybrid.
+
+:::
+
+Use this connectivity type for access to on-premises resources or inside onther cloud providers.
```
Comment (lzdanski) [inline astro/connect-gcp.md:131]:
```suggestion
Use this connectivity type to access on-premises resources or resources in other cloud providers.
```

File: astro/connect-azure.md [hunk 9]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
-1. Retrieve the following information from the target Azure cluster that you want to connect with:
+1. Retrieve the following information from the target Azure environment that you want to connect with:
 
-    - Azure Subscription ID.
+    - Azure Tenant ID and Subscription ID.
     - VNet ID.
     - Resource Group ID.
 
-2. Run the following Azure CLI commands to give Astronomer support temporary permissions to establish a VNet peering connection:
+2. Prepare JSON file `astro-vnet-peering-creator-role.json` with the following permisisons (`{customer-subscription-id}` has to be replaced with your value):
```
Comment (lzdanski) [inline astro/connect-azure.md:92]:
```suggestion
2. Prepare the `astro-vnet-peering-creator-role.json` JSON file with the following permissions. Replace `{customer-subscription-id}` with your value:
```

File: astro/connect-azure.md [hunk 10]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+    --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 \
+    --role "Astro VNET Peering Contributor" \
+    --scope "/subscriptions/{customer-subscription-id}"
+
+    # Verify an assignment
+    az role assignment list --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --all -o table
     ```
 
-3. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you have granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
+4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you have granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```
Comment (lzdanski) [inline astro/connect-azure.md:136]:
```suggestion
4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```

File: astro/connect-azure.md [hunk 11]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
     - Astro Cluster ID 
     - Azure Tenant ID and Subscription ID of the target VNet
-    - VNet ID of the target VNet
+    - Resource group ID
+    - VNet ID for the peering
 
 After receiving your request, Astronomer support creates a VNet peering connection between the two VNets. No other actions are required from you. Astronomer support will notify you when the connection is ready to use.
 
-When the network connection is confirmed, you can delete the temporary roles you created using the [Azure CLI](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-remove#azure-cli).
+When the network connection is confirmed, you can delete the temporary roles you created using command (`{customer-subscription-id}` has to be replaced with your value):
```
Comment (lzdanski) [inline astro/connect-azure.md:145]:
```suggestion
When the network connection is confirmed, you can delete the temporary roles you created using the following command. Replace `{customer-subscription-id}` with your value:
```

File: astro/connect-azure.md [hunk 12]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+    az role assignment delete --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --role "Astro VNET Peering Contributor" --scope "/subscriptions/{customer-subscription-id}"
+    ```
+
+</TabItem>
+
+<TabItem value="VHub peering">
+
+:::info
+
+This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
```
Comment (lzdanski) [inline astro/connect-azure.md:157]:
```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

File: astro/connect-azure.md [hunk 13]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+:::
+
+To set up a private connection between an Astro Virtual Network (VNet) and an Azure VHub, you can create a VHub peering connection. VHub peering ensures private and secure connectivity, reduces network transit costs, and attach Astro environment to centralized managed network.
```
Comment (lzdanski) [inline astro/connect-azure.md:161]:
```suggestion
To set up a private connection between an Astro Virtual Network (VNet) and an Azure VHub, you can create a VHub peering connection. VHub peering ensures private and secure connectivity, reduces network transit costs, and attaches the Astro environment to a centralized managed network.
```

File: astro/connect-azure.md [hunk 14]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+1. Retrieve the following information from the target Azure environment that you want to connect with:
+
+    - Azure Tenant ID and Subscription ID.
+    - VHub name.
+    - Resource Group ID.
+    - Optional. Firewall IP address if you use any on the VHub side.
+
+2. Prepare JSON file `astro-vhub-peering-creator-role.json` with the following permisisons (`{customer-subscription-id}` has to be replaced with your value):
```
Comment (lzdanski) [inline astro/connect-azure.md:170]:
```suggestion
2. Prepare a `astro-vhub-peering-creator-role.json` JSON file with the following permissions. Replace `{customer-subscription-id}` with your value:
```

File: astro/connect-azure.md [hunk 15]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+        "NotActions": [
+      
+        ],
+        "AssignableScopes": [
+          "/subscriptions/{customer-subscription-id}"
+        ]
+    }
+    ```
+
+3. Run the following Azure CLI commands to give Astronomer support temporary permissions to establish a VHub peering connection:
```
Comment (lzdanski) [inline astro/connect-azure.md:193]:
```suggestion
3. Run the following Azure CLI commands to give Astronomer Support temporary permissions to establish a VHub peering connection:
```
Do we want this in public docs?

File: astro/connect-azure.md [hunk 16]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+    az role assignment create \
+    --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 \
+    --role "Astro VHub Peering Contributor" \
+    --scope "/subscriptions/{customer-subscription-id}"
+
+    # Verify an assignment
+    az role assignment list --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --all -o table
+    ```
+
+4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you have granted permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```
Comment (lzdanski) [inline astro/connect-azure.md:212]:
```suggestion
4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you granted them permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
```

File: astro/connect-azure.md [hunk 17]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+    - Astro Cluster ID 
+    - Azure Tenant ID and Subscription ID with a VHub
+    - Resource group ID
+    - VHub name and preferable name for the peering
+    - Optional. Firewall IP address if you use any on the VHub side.
```
Comment (lzdanski) [inline astro/connect-azure.md:218]:
```suggestion
    - (Optional) Firewall IP address if you use any on the VHub side.
```

File: astro/connect-azure.md [hunk 18]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+After receiving your request, Astronomer support creates a VHub peering connection to Astro VNet. No other actions are required from you. Astronomer support will notify you when the connection is ready to use.
+
+When the network connection is confirmed, you can delete the temporary roles you created using command (`{customer-subscription-id}` has to be replaced with your value):
```
Comment (lzdanski) [inline astro/connect-azure.md:222]:
```suggestion
When the network connection is confirmed, you can delete the temporary roles you created using the following command. Replace `{customer-subscription-id}` with your value:
```

File: astro/connect-azure.md [hunk 19]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
```
Comment (lzdanski) [inline astro/connect-azure.md:267]:
```suggestion
This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```

File: astro/connect-azure.md [hunk 20]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
...
+
+:::
+
+Use this connectivity type for access to on-premises resources or inside onther cloud providers.
```
Comment (lzdanski) [inline astro/connect-azure.md:271]:
```suggestion
Use this connectivity type to access on-premises resources or resources in other cloud providers.
```

File: astro/connect-azure.md [hunk 21]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
...
+
+#### Perequisites
+
+Retrieve the following information about your VPN device or application:
+
+    - Public IP address
+    - Subnet CIDR (multipple if needed) on your side of the connection
```
Comment (lzdanski) [inline astro/connect-azure.md:278]:
```suggestion
    - Subnet CIDR range, multiple if needed, for your side of the connection
```

File: astro/connect-azure.md [hunk 22]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
...
+    - Preferences regarding shared key and BGP usage
+    - IKE settings for the tunnel
+
+#### Contact Astronomer support for VPN configuration on Astro side
+
+Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). CRE team will proceed with the required steps. In case of additional questions or clarifications, CRE team will contact you within Support request or let you know about connectivity tests.
```
Comment (lzdanski) [inline astro/connect-azure.md:284]:
```suggestion
Submit all collected details to [Astronomer support](https://cloud.astronomer.io/open-support-request). The Astronomer CRE team will proceed with the required steps. The CRE team will contact you using your support ticket to ask follow-up questions, request clarification, or let you know about connectivity tests.
```

**Review by lzdanski — APPROVED** [review]

**Comment by lzdanski** [comment]
Followed up with @afastronomer to confirm that Rocco + Piotr added as FYI vs essential reviewers

File: astro/connect-aws.md [hunk 23]
```diff
@@ -77,6 +77,7 @@ Choose one of the following setups based on the security requirements of your co
         {label: 'VPC peering', value: 'VPC peering'},
         {label: 'Transit Gateways', value: 'Transit Gateways'},
         {label: 'AWS PrivateLink', value: 'AWS PrivateLink'},
+        {label: 'VPN', value: 'VPN'},
```
Comment (chomipi88) [inline astro/connect-aws.md:80]: Need to add this option to the list on line 40

File: astro/connect-azure.md [hunk 24]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+    az role assignment delete --assignee a67e6057-7138-4f78-bbaf-fd9db7b8aab0 --role "Astro VNET Peering Contributor" --scope "/subscriptions/{customer-subscription-id}"
+    ```
+
+</TabItem>
+
+<TabItem value="VHub peering">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```
Comment (chomipi88) [inline astro/connect-azure.md:157]: The text here is different from the other info banners

File: astro/connect-aws.md [hunk 25]
```diff
@@ -294,6 +295,36 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```
Comment (chomipi88) [inline astro/connect-aws.md:302]: The text here is different from the other info banners

File: astro/connect-azure.md [hunk 26]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```
Comment (chomipi88) [inline astro/connect-azure.md:267]: The text here is different from the other info banners

File: astro/connect-azure.md [hunk 27]
```diff
@@ -68,7 +68,9 @@ The option that you choose is determined by the security requirements of your co
     groupId="create-a-private-connection-between-astro-and-azures"
     values={[
         {label: 'VNet peering', value: 'VNet peering'},
+        {label: 'VHub peering', value: 'VHub peering'},
```
Comment (chomipi88) [inline astro/connect-azure.md:71]: add to list of options on line 33

File: astro/connect-azure.md [hunk 28]
```diff
@@ -68,7 +68,9 @@ The option that you choose is determined by the security requirements of your co
...
         {label: 'Azure Private Link', value: 'Azure Private Link'},
+        {label: 'VPN', value: 'VPN'},
```
Comment (chomipi88) [inline astro/connect-azure.md:73]: add to list of options on line 33

File: astro/connect-azure.md [hunk 29]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+4. Contact [Astronomer support](https://cloud.astronomer.io/open-support-request) to tell them that you granted them permissions to the Astronomer Service Principal. In addition, provide the following details in your request:
+
+    - Astro Cluster ID 
+    - Azure Tenant ID and Subscription ID with a VHub
+    - Resource group ID
+    - VHub name and preferable name for the peering
+    - (Optional) Firewall IP address if you use any on the VHub side.
+
+After receiving your request, Astronomer support creates a VHub peering connection to Astro VNet. No other actions are required from you. Astronomer support will notify you when the connection is ready to use.
```
Comment (chomipi88) [inline astro/connect-azure.md:220]: @afastronomer should this say from Astro VNet? Not to Astro VNet.

File: astro/connect-gcp.md [hunk 30]
```diff
@@ -69,6 +69,7 @@ Choose one of the following setups based on the security requirements of your co
     values={[
         {label: 'VPC peering', value: 'vpc'},
         {label: 'Private Service Connect', value: 'Private-Service-Connect'},
+        {label: 'VPN', value: 'VPN'},
```
Comment (chomipi88) [inline astro/connect-gcp.md:72]: add option to list above

File: astro/connect-gcp.md [hunk 31]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
```
Comment (chomipi88) [inline astro/connect-gcp.md:127]: The text here is different from the other info banners

**Review by chomipi88 — APPROVED** [review]
few small comments. lgtm.

File: astro/connect-azure.md (hunk 29 above)
Comment (afastronomer) [reply astro/connect-azure.md:220]: When you do this type of the peering, every time you connect from Hub to Spokes. We login into customer environment and **from there** we initiate a connection.

Comment (chomipi88) [reply astro/connect-azure.md:220]: Ah got it thanks.

File: astro/connect-aws.md [hunk 32]
```diff
@@ -294,6 +295,36 @@ You'll incur additional AWS infrastructure costs for every AWS PrivateLink endpo
...
+
+:::
```
Comment (lzdanski) [inline astro/connect-aws.md:304]:
```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

File: astro/connect-azure.md [hunk 33]
```diff
@@ -81,44 +83,147 @@ This connection option is only available for dedicated Astro Hosted clusters and
...
+
+</TabItem>
+
+<TabItem value="VHub peering">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
+
+:::
```
Comment (lzdanski) [inline astro/connect-azure.md:159]:
```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

File: astro/connect-azure.md [hunk 34]
```diff
@@ -155,6 +260,31 @@ Note that you'll incur additional Azure infrastructure costs for every Azure pri
 
 </TabItem>
 
+<TabItem value="VPN">
+
+:::info
+
+This connection option is not available for standard Astro Hosted clusters. It is available for dedicated clusters and Astro Hybrid.
+
+:::
```
Comment (lzdanski) [inline astro/connect-azure.md:269]:
```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

File: astro/connect-gcp.md [hunk 35]
```diff
@@ -119,6 +120,31 @@ If you don't see your service listed, open a support case with [Astronomer suppo
...
+
+:::
```
Comment (lzdanski) [inline astro/connect-gcp.md:129]:
```suggestion
:::info
This connection option is only available for dedicated Astro Hosted clusters and Astro Hybrid.
:::
```

**Pull request merged**